AirCasting/SessionViews/ShareSession/ShareSessionViewModel.swift

```diff
@@ -5,36 +5,45 @@ import Foundation
 
 enum ShareSessionError: Error {
     case noSessionURL
+    case requestError
 }
 
 protocol ShareSessionViewModel: ObservableObject {
     var streamOptions: [ShareSessionStreamOptionViewModel] { get set }
     var alert: AlertInfo? { get set }
     var showShareSheet: Bool { get set }
+    var showInvalidEmailError: Bool { get set }
     var sharingLink: URL? { get set }
+    var email: String { get set }
     func didSelect(option: ShareSessionStreamOptionViewModel)
-    func shareLinkTepped()
+    func shareLinkTapped()
+    func shareEmailTapped()
     func cancelTapped()
     func sharingFinished()
+    func isEmailValid() -> Bool
 }
 
 class DefaultShareSessionViewModel: ShareSessionViewModel {
     @Published var alert: AlertInfo?
     @Published var showShareSheet: Bool = false
+    @Published var showInvalidEmailError: Bool = false
     @Published var sharingLink: URL?
-    private let exitRoute: () -> Void
+    @Published var email: String = ""
+    private let exitRoute: (Bool?) -> Void
```

Comment:
looking at the rest of the code this `exitRoute` only gets called with `true` or `nil`. It's a bit confusing. I'd suggest creating an enum for this instead of using `Bool`. It could look something like this:

```Swift
enum ShareSessionResult {
    case sessionShared
    case cancelled
    ... etc ...
}
```